### PR TITLE
Remove deprecated check/service fields and metric names

### LINKED ISSUE #4033: API key name cleanup / deprecations meta issue
We have a number of planned changes and deprecations in API responses, specifically for key names. This issue is to track them in a central place to ensure they get into the next major release as breaking changes. These changes already support the alternative, we are just removing the legacy option. These changes would affect both the APIs and any static configuration.

- [ ] Check definition: `CheckID`will be replaced with `id`
- [ ] Service definition: `enableTagOverride` will be replaced with `enable_tag_override`
- [ ] Check definition: `script` will be replaced with `args`
- [ ] Telemetry: `consul.consul` metric prefix for backwards compatibility will be removed, only `consul.` will be sent, along with the removal of `enable_deprecated_names`
- [ ] Auto-join: Ensure we've swept up all instances of `retry-join-<provider>` from everywhere

It is possible to delay some of these changes if we consider them to be too impactful to users.

### PR DESCRIPTION
**Pull request by kyhavlov**
This removes previously deprecated fields from check (`CheckID`, `script`) and service (`enableTagOverride`) definitions and removes the old `consul.consul...` style metric names along with the `enable_deprecated_names` metrics config option.

Closes #4033.

### CODE REVIEW

**Comment by pearkes** [comment]
Can we also add a section for 1.1.0 and detail these changes on the [upgrade guide page](https://www.consul.io/docs/upgrade-specific.html)? I believe we previously talked about the deprecations but given the removal it should be detailed there.

File: website/source/docs/upgrade-specific.html.md [hunk 1]
```diff
@@ -14,6 +14,18 @@ details provided for their upgrades as a result of new features or changed
 behavior. This page is used to document those details separately from the
 standard upgrade flow.
 
+## Consul 1.1.0
+
+### Removal of deprecated features
```
Comment (banks) [inline website/source/docs/upgrade-specific.html.md]: Nitpick: seems like we Title Case headings below.

File: website/source/docs/upgrade-specific.html.md [hunk 2]
```diff
@@ -14,6 +14,18 @@ details provided for their upgrades as a result of new features or changed
...
+
+The following previously deprecated fields and config options have been removed:
+
+ - `CheckID` has been removed from config file check definitions (use `id` instead).
+ - `script` has been removed from config file check definitions (use `args` instead).
+ - `enableTagOverride` is no longer valid in service definitions (use `enable_tag_override` instead).
+ - The deprecated set of metric names (beginning with `consul.consul.`) has been removed
+ along with the `enable_deprecated_names` option from the metrics configuration.
+
```
Comment (banks) [inline website/source/docs/upgrade-specific.html.md]: Should we link to the changelog/update guide that deprecated them in the first place to highlight how long people this surprises have had to notice?

**Comment by pearkes** [comment]
Agree with Paul's nits but otherwise upgrade instructions LGTM!

**Pull request merged**